src/image-viewer.js

```diff
@@ -341,28 +333,72 @@ export class ImageViewer {
       const deltaY = this.viewerBox_.height / 2 - e.data.clientY;
       this.onZoom_(newScale, deltaX, deltaY, true).then(() => {
         return this.onZoomRelease_();
-      });
+      }).then(() => this.updatePanningGestureRegistration_());
     });
-    gestures.onGesture(TapzoomRecognizer, e => {
+
+    this.gestures_.onGesture(TapzoomRecognizer, e => {
       event.preventDefault();
       this.onTapZoom_(e.data.centerClientX, e.data.centerClientY,
           e.data.deltaX, e.data.deltaY);
       if (e.data.last) {
         this.onTapZoomRelease_(e.data.centerClientX, e.data.centerClientY,
-            e.data.deltaX, e.data.deltaY, e.data.velocityY, e.data.velocityY);
+            e.data.deltaX, e.data.deltaY, e.data.velocityY, e.data.velocityY)
+            .then(() => this.updatePanningGestureRegistration_());;
       }
     });
-    gestures.onGesture(PinchRecognizer, e => {
+
+    this.gestures_.onGesture(PinchRecognizer, e => {
       // To disable iOS 10 Safari default pinch zoom
       event.preventDefault();
       this.onPinchZoom_(e.data.centerClientX, e.data.centerClientY,
           e.data.deltaX, e.data.deltaY, e.data.dir);
       if (e.data.last) {
-        this.onZoomRelease_();
+        this.onZoomRelease_().then(() => {
+          this.updatePanningGestureRegistration_();
+        });;
       }
     });
   }
 
+/**
```

Comment:
nit: this comment should be at the same indentation level as the function

Reply:
Removing this function in refactor.